kinova_driver/src/kinova_arm.cpp

```diff
@@ -561,16 +574,8 @@ void KinovaArm::publishJointAngles(void)
     joint_state.velocity[2] = current_vels.Actuator3;
     joint_state.velocity[3] = current_vels.Actuator4;
     // no velocity info for fingers
-    if(finger_number_==2)
-    {
-        joint_state.velocity[joint_total_number_-2] = 0;
-        joint_state.velocity[joint_total_number_-1] = 0;
-    }
-    else if(finger_number_==3)
-    {
-        joint_state.velocity[joint_total_number_-3] = 0;
-        joint_state.velocity[joint_total_number_-2] = 0;
-        joint_state.velocity[joint_total_number_-1] = 0;
+    for(int fi=0; fi<(2*finger_number_); fi++) {
```

Comment:
This should be :
```for(int fi=1; fi<=(2*finger_number_); fi++) {```

Reply:
I think I got confused by the backward indexing. I think that should also be 1 ... (2*fingers)-1 because otherwise we would access `joint_state.velocity[joint_total_number_]`, which eventually leads to a SEGFAULT.

I changed this to:
```C++
for(int fi=arm_joint_number_; fi<joint_total_number_; fi++) {
    joint_state.velocity[fi] = 0;
}
```
It explcitely iterates from the last arm joint index to the last finger joint index.

IMHO, this is much clearer which indices are accessed.

Reply:
Definitely!